sklearn/feature_selection/_from_model.py

```diff
@@ -356,9 +377,16 @@ def partial_fit(self, X, y=None, **fit_params):
         self : object
             Fitted estimator.
         """
-        if self.prefit:
-            raise NotFittedError("Since 'prefit=True', call transform directly")
-        if not hasattr(self, "estimator_"):
+        if self.prefit and not hasattr(self, "estimator_"):
+            try:
+                check_is_fitted(self.estimator)
+            except NotFittedError as exc:
+                raise NotFittedError(
+                    "When `prefit=True`, `estimator` is expected to be a fitted "
+                    "estimator."
+                ) from exc
+            self.estimator_ = deepcopy(self.estimator)
+        elif not hasattr(self, "estimator_"):
```

Comment:
Does it really make sense to allow to partial_fit when `prefit=True` ?
To me allowing `fit` makes sense (and should probably be mandatory) but it does not fit the estimator if prefit=True, it only does some validations.
Here, when prefit=True, the estimator still passes through partial_fit which is kind of unexpected

Reply:
We can prevent a `partial_fit`. I will document it and add a subsequent test.